### ISSUE TITLE
enhance twitchat accessibility

### ISSUE THREAD

**Issue by yplassiard** [Durss/Twitchat#5]
Although this interface looks very promising, it lacks some features to be really usable by people with disabilities, and especially blind people using screen reader.
Here is a list of issues I found regarding this topic:
- All buttons should be labelled with what they open/do (settings for the settings button for example)
- - Within settings, there is no way to know if a box (a feature) is enabled or not: Either use a html checkbox to represent the checkbox, or, if that's not possible, you may add a "disabled" or "enabled" text alongside the feature's title, so that the information is rendered.
- - When a chat message arrives, could be nice to use a aria-live region to read it. A way to do this is using a <p role="alert"></p> and when an event occurs (such as a chat message), adds the corresponding message within this tag for about a few seconds to let the screen reader read it. after the message is read, the <P> tag can be reset to an empty content.

I'll try to see if I can create one/more PRs to enhance this. If so, they will be marked as "a11y" as it concerns accessibility.

**Comment by Durss** [comment]
Thank you so much for your feedbacks !
I have no accessibility knowledges, I really appreciateyou take time to share some !
I'll definitely try to solve these problems !

**Comment by Durss** [comment]
Just to make sure i do things right, for your first point "all buttons should be labeled", does an "aria-label" attribute would do the job ?

**Comment by yplassiard** [comment]
Sorry for getting back to you with such delay!
Whenever possible, using an alternate text such the one you mentionned within the button tag is a good idea.
